### PR TITLE
Dlg replace

### PR DESCRIPTION
**Pull request by stevenndungu**

### CODE REVIEW

**Comment by stevenndungu** [comment]
@africanmathsinitiative/developers  this is ready for review.
Note: @dannyparsons, Just correction  from what I had written on  issue #1779 on na.locf function ` "if the first or the last value of the vector is missing ,it reverses using the opposite approach to fill up the missing value which is a plus for this over fill"` ...  While implementing I have seen that that is not the case, what happens is that na.rm is TRUE by default it thus truncates missing values not replaced by the applied aproach. It thus works in a similar case as fill.

This is what i mean.

We can use this data set for testing:
[DATA_MISSING.xlsx](https://github.com/africanmathsinitiative/R-Instat/files/1040636/DATA_MISSING.xlsx)
`> bz <- c(NA,NA,NA,4,NA,NA,5,NA,5,NA,NA,NA)
> bz
 [1] NA NA NA  4 NA NA  5 NA  5 NA NA NA
> na.locf(bz)
[1] 4 4 4 5 5 5 5 5 5
> na.locf(bz, fromLast = TRUE,na.rm=F)
 [1]  4  4  4  4  5  5  5  5  5 NA NA NA
> na.locf(bz, fromLast = FALSE,na.rm=F)
 [1] NA NA NA  4  4  4  5  5  5  5  5  5`

Everything is working well as it should be. and the dialogue is ready for testing

**Comment by dannyparsons** [comment]
I think that's fine and we are not as interested in that functionality anyway. We don't want it to do the truncating so as long as we always use `na.rm = FALSE` it should be fine.

**Comment by stevenndungu** [comment]
Great, I actually set `na.rm = FALSE` on the function.

**Comment by lilyclements** [comment]
Reviewing/making amendments to this now

**Comment by stevenndungu** [comment]
Point 1: Not sure when that is happening exactly. Will skype you so that I can be able to see that.
Point 2: I made a mistake on the rdo to be checked when oldmissing rdo is checked. Thanks for noticing that.
Point 3: You are the one who implemented that, so I did not tamper with that since I thought you had a good reason of doing that. Check line 213 of the dialogue

**Comment by stevenndungu** [comment]
@lilyclements  I have fixed point 2

**Comment by lilyclements** [comment]
Point 1: If you open the dialog, just put a variable in the receiver and press OK (for the first time opening the dialog) then there is an error:
>   argument "new_value" is missing, with no default


Point 2: Great, good job for fixing


Point 3: I knew you didn't cause that issue, it was just one I noticed when using it.
Great, thanks. I remember why we've implemented that now, you can do an interval with a factor. Makes perfect sense.

Just point 1 then it should be ready!

**Comment by lilyclements** [comment]
@dannyparsons this is ready to review

**Comment by stevenndungu** [comment]
@dannyparsons  this is now conflict free.

File: instat/static/InstatObject/R/data_object_R6.R [hunk 1]
```diff
@@ -601,103 +602,119 @@ data_object$set("public", "replace_value_in_data", function(col_names, rows, old
...
     }
     if(!done) {
-      replace_rows[is.na(replace_rows)] <- FALSE
-      if(sum(replace_rows) > 0) {
-        if(filter_applied) {
-          replace_rows <- replace_rows & curr_filter
+      if(locf){
+        
+        print(nrow(private$data))
+        print(length(my_data))
```
Comment (dannyparsons) [inline instat/static/InstatObject/R/data_object_R6.R]: can you remove these prints

**Comment by dannyparsons** [comment]
Is this done now?

**Comment by stevenndungu** [comment]
Nop, still has one developer error. Figuring out how it comes about and why?This  happens when you run  either of last radio buttons, check misssing radio button, run and reopen the dialogue. The developer error occurrs.Else other things are okay.

**Comment by dannyparsons** [comment]
@stevenndungu here I fixed the issue https://github.com/africanmathsinitiative/R-Instat/pull/3281/commits/6188aa3cc82d2c8f9b749d3e9452d235669f5238
The `locf` and `from_last` parameters we not being removed when you unchecked from above/below, so the conditions were never satisfied for other radio buttons.
Also I noticed From Above/Below were the wrong buttons and I changed the text of the buttons.

**Comment by stevenndungu** [comment]
Aaah! I see, I was wondering why there was an issue. Thanks for helping in sorting that out. I

**Pull request merged**